File: frontend/apps/web-antd/src/api/aas.ts
/**
 * AAS API — 位号元数据同步与连接配置.
 *
 * 对接后端 /api/v1/aas/*（schemas/aas.py）；同步为异步 Celery 任务，
 * 经 GET /aas/config 的 lastSyncStatus 轮询进度（PROCESSING → SUCCESS/FAILED）。
 */
import { requestClient } from '#/api/request';

export namespace AasApi {
  /** AAS 连接配置与最近同步状态（GET /aas/config，仅 ADMIN） */
  export interface AasConfig {
    /** OPC UA 端点 URL */
    endpoint: string;
    /** 同步周期（秒，定时同步已停用，仅展示） */
    syncIntervalSeconds: number;
    /** 是否启用定时同步 */
    enabled: boolean;
    /** 是否为 Mock 模式（无真实 AAS） */
    mockMode: boolean;
    /** 安全模式：None/Sign/SignAndEncrypt */
    securityMode: string;
    /** 最近一次同步完成时间（ISO 8601），未同步过为 null */
    lastSyncAt: null | string;
    /** 最近一次同步状态：PROCESSING/SUCCESS/FAILED，未同步过为 null */
    lastSyncStatus: 'FAILED' | 'PROCESSING' | 'SUCCESS' | null;
  }

  /** 同步触发结果（POST /aas/sync） */
  export interface SyncTriggerResult {
    /** Celery 任务 ID */
    taskId: string;
    /** 触发后立即返回的状态（恒 PROCESSING） */
    status: string;
    /** 任务进度查询地址 */
    checkUrl: null | string;
  }
}

/** 获取 AAS 连接配置与最近同步状态（仅 ADMIN） */
export function getAasConfigApi() {
  return requestClient.get<AasApi.AasConfig>('/aas/config');
}

/** 手动触发 AAS 位号同步（异步任务，经 getAasConfigApi 轮询 lastSyncStatus） */
export function triggerAasSyncApi() {
  return requestClient.post<AasApi.SyncTriggerResult>('/aas/sync');
}

File: backend/app/schemas/aas.py
"""AAS-related schemas (IDS v3.2 §2.2.5~2.2.6, §3.2.1)."""

from __future__ import annotations

from pydantic import Field

from app.schemas.base import CamelModel


class AasConfigInfo(CamelModel):
    """AAS 连接配置信息。"""

    endpoint: str = Field(..., description="OPC UA 端点 URL")
    syncIntervalSeconds: int = Field(..., description="同步周期（秒）")
    enabled: bool = Field(..., description="是否启用定时同步")
    mockMode: bool = Field(..., description="是否为 Mock 模式（无真实 AAS）")
    securityMode: str = Field("None", description="安全模式：None/Sign/SignAndEncrypt")


class AasConfigUpdate(CamelModel):
    """PUT /api/v1/aas/config 请求体。"""

    endpoint: str | None = Field(None, description="OPC UA 端点 URL")
    syncIntervalSeconds: int | None = Field(None, ge=30, le=86400, description="同步周期（秒）")
    enabled: bool | None = Field(None, description="是否启用定时同步")
    securityMode: str | None = Field(None, description="安全模式：None/Sign/SignAndEncrypt")


class AasConfigTestResult(CamelModel):
    """POST /api/v1/aas/config/test 响应。"""

    success: bool
    latencyMs: int | None = None
    message: str


class AasSyncTriggerResult(CamelModel):
    """POST /api/v1/aas/sync 响应。"""

    taskId: str
    status: str = "PROCESSING"
    checkUrl: str | None = None


class AasTagItem(CamelModel):
    """AAS Tag 列表项。"""

    tagId: str
    tagName: str
    description: str | None = None
    tagType: str | None = None
    currentValue: float | None = None
    quality: str | None = None
    lastSyncAt: str | None = None
    isLinked: bool = False
    associatedLoopId: str | None = None
    associatedLoopTagName: str | None = None


class AasTagListData(CamelModel):
    """AAS Tag 列表响应 data 块。"""

    items: list[AasTagItem]
    total: int
    page: int
    pageSize: int
    lastSyncAt: str | None = None
    syncStatus: str = "SUCCESS"


__all__ = [
    "AasConfigInfo",
    "AasConfigTestResult",
    "AasConfigUpdate",
    "AasSyncTriggerResult",
    "AasTagItem",
    "AasTagListData",
]
